### PR TITLE
Update CLI to automatically geosearch collection locations from CSVs

### PR DESCRIPTION
**Pull request by jshoe**
## Description
- Mirrors the update to the web to automatically geosearch collection locations from CSV uploads.
- Search for all raw_names -> Ask user to accept/reject unique matches -> Display table of final matches
- fetch_location_matches will do max 5 search requests at a time with up to 3 tries.

### Bulk mode
Bulk mode

### Single sample
Single sample

### Python 2
Python 2

### Search error timeout
Search error

### Web confirmation
On web

### CODE REVIEW

**Comment by jshoe** [comment]
^ updated the "~" to "(!)" and tweaked wording based on Jenn's comments

File: idseq/uploader.py [hunk 1]
```diff
@@ -177,14 +178,19 @@ def upload(sample_name, project_id, headers, url, r1, r2, chunk_size, csv_metada
...
 
-    resp = requests.post(
+    raw_resp = requests.post(
         url + '/samples/bulk_upload_with_metadata.json', data=json.dumps(data), headers=headers)
-    resp = resp.json()
+    resp = raw_resp.json()
 
-    if len(resp.get("errors", {})) == 0:
-        print("Connected to the server.")
+    if raw_resp.status_code == 200:
```
Comment (jshoe) [inline idseq/uploader.py]: This fixes the "deprecation warning", which doesn't have an "errors" field

File: idseq/locations.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    set_location_matches(csv_data, matched_locations)
+    print_location_matches(csv_data, base_url, project_id)
+    return csv_data
+
+
+def get_raw_locations(csv_data):
+    raw_names = set()
+    for metadata in csv_data.values():
+        for field_name, value in metadata.items():
+            if field_name.lower() in ["collection_location", "collection location"]:
```
Comment (tfrcarvalho) [inline idseq/locations.py]: I wonder if we should have this fields from server?
Maybe add it to a visible constant on top at least ( I see you repeat it further down)?

File: idseq/locations.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+
+def set_location_matches(csv_data, matched_locations):
+    for sample_name, metadata in csv_data.items():
+        for field_name, value in metadata.items():
+            if field_name.lower() in ["collection_location", "collection location"]:
+                if value in matched_locations:
+                    result = matched_locations[value]
+                    is_human = (
+                        metadata.get("host_genome") or metadata.get("Host Genome")
```
Comment (tfrcarvalho) [inline idseq/locations.py]: Should we use lower here too ?

File: idseq/locations.py (hunk 2 above)
Comment (jshoe) [reply idseq/locations.py]: 👍

File: idseq/locations.py (hunk 3 above)
Comment (jshoe) [reply idseq/locations.py]: 👍

**Pull request merged**